src/status_im2/contexts/onboarding/sign_in/view.cljs

```diff
@@ -1,42 +1,223 @@
+  [:<>
+   [rn/view {:style style/scan-qr-code-container}]
+   [calculate-qr-scan-hole-area qr-view-finder]
+   (if @camera-permission-granted?
+     [viewfinder @qr-view-finder]
+     [enable-camera-access-view request-camera-permission])])
+
+(defn- enter-sync-code-tab
   []
-  [rn/view {:style style/page-container}
-   [navigation-bar]
-   [rn/view {:style {:padding-horizontal 20}}
+  [rn/view {:style style/enter-sync-code-container}
+   [quo/text
+    {:size   :paragraph-1
+     :weight :medium
+     :style  {:color colors/white}}
+    "Yet to be implemented"]])
+
+(defn- bottom-view
+  [insets]
+  [rn/touchable-without-feedback
+   {:on-press #(js/alert "Yet to be implemented")}
+   [rn/view
+    {:style (style/bottom-container (:bottom insets))}
     [quo/text
-     {:size   :heading-1
-      :weight :semi-bold
-      :style  {:color colors/white}} "Sign in by syncing"]
-    [quo/button
-     {:on-press #(rf/dispatch [:navigate-to :syncing-devices])
-      :style    {}} (i18n/label :t/continue)]]])
+     {:size   :paragraph-2
+      :weight :regular
+      :style  style/bottom-text}
+     (i18n/label :t/i-dont-have-status-on-another-device)]]])
+
+(defn- check-qr-code-data
+  [event]
+  (when-let [data (string/trim (oops/oget event "nativeEvent.codeStringValue"))]
+    (if (string/starts-with? data constants/local-pairing-connection-string-identifier)
+      (rf/dispatch [:syncing/input-connection-string-for-bootstrapping data])
+      (rf/dispatch [:toasts/upsert
+                    {:icon       :info
+                     :icon-color colors/danger-50
+                     :text       (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
 
-(defn sign-in
+(defn view
   []
-  [rn/view {:style {:flex 1}}
-   [background/view true]
-   [page]])
+  (let [active-tab                 (reagent/atom 1)
+        camera-permission-granted? (reagent/atom false)
+        qr-view-finder             (reagent/atom {})
+        height                     (rf/sub [:dimensions/window-height])
+        width                      (rf/sub [:dimensions/window-width])]
+    [:f>
+     (fn []
+       (let [insets                    (safe-area/use-safe-area)
+             show-holes?               (and @qr-view-finder
+                                            (= @active-tab 1)
+                                            @camera-permission-granted?)
+             holes                     (merge @qr-view-finder {:borderRadius 16})
+             show-camera?              (and (= @active-tab 1) @camera-permission-granted?)
+             camera-ref                (atom nil)
+             read-qr-once?             (atom false)
+             on-read-code              (fn [data]
+                                         (when-not @read-qr-once?
+                                           (reset! read-qr-once? true)
+                                           (check-qr-code-data data)))
+             request-camera-permission (fn []
```

Comment:
You can move `request-camera-permission` to the outside `let` to reduce the overhead in the renderer.